classes/converter_chromium.php

```diff
@@ -117,6 +121,51 @@ protected function generate_pdf_content(moodle_url $proxyurl, string $filename =
         }
     }
 
+    /**
+     * Checks if MathJax is defined on the page, and waits for it to finish processing LaTeX equations.
+     * If MathJax is not defined, the function assumes there is no MathJax rendering and proceeds.
+     *
+     * The check is performed in a loop, with a 0.5 second pause between checks, until the MathJax queue
+     * is empty or the timeout is reached.
+     *
+     * @param  Page $page The current browser page.
+     * @param  int $timeout The maximum time to wait for the MathJax to finish processing, in milliseconds.
+     *                      Defaults to 30000ms (30 seconds).
+     * @throws \Exception If MathJax does not finish rendering within the specified timeout.
+     */
+    protected function wait_for_mathjax_condition(Page $page, int $timeout = 30000): void {
```

Comment:
It feels a bit odd to have this function that's specific to mathjax. Is there anyway it could be implemented more generally? Like `wait_for_javascript` or something like that?

Reply:
I pushed a fixup commit here. Updated it to be `wait_for_js_condition`  https://github.com/catalyst/moodle-tool_pdfpages/pull/40/commits/aaac411fe1e289664771b5bb50e7b6789085daa8 

PS: I also pushed some commits to for improvements

Reply:
Hey @tsmilan it seems like this might be causing issues with the **test.php** script. When I try this: http://localhost/mymoodle/admin/tool/pdfpages/test.php?url=http://localhost/mymoodle/course/view.php?id=3 I'm getting an error:

Exception - tool_pdfpages\converter_chromium::wait_for_js_condition(): Argument #3 ($jsconditionparams) must be of type array, null given, called in [dirroot]/admin/tool/pdfpages/classes/converter_chromium.php on line 115

Reply:
Hi Cam, I pushed a fix here https://github.com/catalyst/moodle-tool_pdfpages/pull/40/commits/d4f67bd153ca6ac7e9696307decf64f2235ec002

You might need to disable shm if you're using a docker container as mentioned here https://github.com/chrome-php/chrome/issues/446#issuecomment-1307304445 to avoid the operation timed out error after 30s
```
    protected function generate_pdf_content(moodle_url $proxyurl, string $filename = '', array $options = [],
                                            string $cookiename = '', string $cookievalue = '',
                                            array $renderoptions = []): string {
        try {
            $browseroptions = [
                'headless' => true,
                'noSandbox' => true,
                'customFlags' => [
                    '--disable-dev-shm-usage',
                ],
            ];
```

Reply:
Test page is working for me now :)